src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,93 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int capacity;
+    private float loadFactor;
+    private Node<K, V>[] buckets;
+
+    public MyHashMap() {
+        this(DEFAULT_CAPACITY, LOAD_FACTOR);
+    }
+
+    public MyHashMap(int capacity, float loadFactor) {
+        this.capacity = capacity;
+        this.loadFactor = loadFactor;
+        this.buckets = new Node[capacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (needsResize()) {
+            resize();
+        }
+        int bucketIndex = getBucketIndex(key);
+        Node<K, V> node = buckets[bucketIndex];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+        Node<K, V> newNode = new Node<>(key, value, buckets[bucketIndex]);
+        buckets[bucketIndex] = newNode;
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int bucketIndex = getBucketIndex(key);
+        Node<K, V> node = buckets[bucketIndex];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getBucketIndex(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode() % capacity);
+    }
+
+    private boolean needsResize() {
+        return (float) size / capacity > LOAD_FACTOR;
+    }
+
+    private void resize() {
+        if (size + 1 > buckets.length * LOAD_FACTOR) {
```

Comment:
why do you need this if when you have method needsResize()?
let's refactor code so that we don't check size 2 times